api/files/serializers.py

```diff
@@ -188,8 +188,12 @@ def user_id(self, obj):
 
     def update(self, instance, validated_data):
         assert isinstance(instance, FileNode), 'Instance must be a FileNode'
-        for attr, value in validated_data.items():
-            setattr(instance, attr, value)
+        if len(validated_data) == 1 and 'checkout' in validated_data:
```

Comment:
Hmmm. I'm having trouble seeing what you're saying in 2. Maybe whiteboarding would help?

And I agree that it's not a problem right now, but it's not an unlikely problem.